.changes/nextrelease/feature-upload-once.json

```diff
@@ -0,0 +1,7 @@
+[
+  {
+    "type": "feature",
+    "category": "S3",
+    "description": "Add Transfer option to upload only non existing files"
```

Comment:
Indeed, I'm not sure about the point to mix versioning and no overwrite. That said, I think this option should take precedence. As the customer needs to change the default overwrite option (from true to false), there won't be any unexpected behavior. If you agree, I can clarify the documentation (for instance, adding "(...) even when versionning enabled").

Reply:
I agree, and I don't think anyone would likely intentionally mix versioning and overwrite, I'd more see it as a potential point of confusion for customers who enable one, forget about it, and later enable the other.  So I wanted your thoughts on how to best clarify that; putting it in the documentation is probably the best option

Reply:
I have updated the option documentation in the code. If this gets merged, it could be good to update the developers guide as well (https://github.com/awsdocs/aws-php-developers-guide/blob/master/doc_source/s3-transfer.rst).